### ISSUE TITLE
Platform name error after update. 

### ISSUE THREAD

**Issue by nzbullet** [beele/homebridge-unifi-protect-camera-motion#36]
I'm receiving an error due to the platform name...Initially I thought it's because that setting has changed and those with existing config need to update and use the UI Settings panel, but even upon changing it I still receive the error. 

`[5/13/2020, 12:05:56] Error loading platform requested in your config.json at position 5
[5/13/2020, 12:05:56] Error: The requested platform 'UnifiProtectMotion' was not registered by any plugin.
    at PluginManager.getPluginForPlatform (/usr/local/lib/node_modules/homebridge/src/pluginManager.ts:223:15)
    at /usr/local/lib/node_modules/homebridge/src/server.ts:386:37
    at Array.forEach (<anonymous>)
    at Server.loadPlatforms (/usr/local/lib/node_modules/homebridge/src/server.ts:373:27)
    at Server.start (/usr/local/lib/node_modules/homebridge/src/server.ts:152:29)
    at cli (/usr/local/lib/node_modules/homebridge/src/cli.ts:78:10)
    at Object.<anonymous> (/usr/local/lib/node_modules/homebridge/bin/homebridge:17:22)
    at Module._compile (internal/modules/cjs/loader.js:955:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:991:10)
    at Module.load (internal/modules/cjs/loader.js:811:32)
    at Function.Module._load (internal/modules/cjs/loader.js:723:14)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1043:10)
    at internal/main/run_main_module.js:17:11
[5/13/2020, 12:05:56] Got SIGTERM, shutting down Homebridge...
homebridge.service: Main process exited, code=exited, status=143/n/a
homebridge.service: Failed with result 'exit-code'.`

**Comment by hfagelnour** [comment]
@beele I Guess I am having the same error after updating the plugin thru the master newly published one (Beta5 was working fine) I also already updated the FFMPG plugin to 1.0.0
please find error below:

[5/13/2020, 2:56:49 AM] ====================
[5/13/2020, 2:56:49 AM] ERROR LOADING PLUGIN homebridge-unifi-protect-camera-motion:
[5/13/2020, 2:56:49 AM] Error: The Node.js native addon module (tfjs_binding.node) can not be found at path: /usr/lib/node_modules/homebridge-unifi-protect-camera-motion/node_modules/@tensorflow/tfjs-node/lib/napi-v5/tfjs_binding.node. 
Please run command 'npm rebuild @tensorflow/tfjs-node build-addon-from-source' to rebuild the native addon module. 
If you have problem with building the addon module, please check https://github.com/tensorflow/tfjs/blob/master/tfjs-node/WINDOWS_TROUBLESHOOTING.md or file an issue.
    at Object.<anonymous> (/usr/lib/node_modules/homebridge-unifi-protect-camera-motion/node_modules/@tensorflow/tfjs-node/dist/index.js:49:11)
    at Module._compile (internal/modules/cjs/loader.js:1156:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1176:10)
    at Module.load (internal/modules/cjs/loader.js:1000:32)
    at Function.Module._load (internal/modules/cjs/loader.js:899:14)
    at Module.require (internal/modules/cjs/loader.js:1042:19)
    at require (internal/modules/cjs/helpers.js:77:18)
    at Object.<anonymous> (/usr/lib/node_modules/homebridge-unifi-protect-camera-motion/src/coco/coco.ts:18:1)
    at Module._compile (internal/modules/cjs/loader.js:1156:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1176:10)
    at Module.load (internal/modules/cjs/loader.js:1000:32)
    at Function.Module._load (internal/modules/cjs/loader.js:899:14)
    at Module.require (internal/modules/cjs/loader.js:1042:19)
    at require (internal/modules/cjs/helpers.js:77:18)
    at Object.<anonymous> (/usr/lib/node_modules/homebridge-unifi-protect-camera-motion/src/coco/loader.ts:2:1)
    at Module._compile (internal/modules/cjs/loader.js:1156:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1176:10)
    at Module.load (internal/modules/cjs/loader.js:1000:32)
    at Function.Module._load (internal/modules/cjs/loader.js:899:14)
    at Module.require (internal/modules/cjs/loader.js:1042:19)
    at require (internal/modules/cjs/helpers.js:77:18)
    at Object.<anonymous> (/usr/lib/node_modules/homebridge-unifi-protect-camera-motion/src/motion/motion.ts:2:1)
[5/13/2020, 2:56:49 AM] ====================
[5/13/2020, 2:56:49 AM] Error loading platform requested in your config.json at position 5
[5/13/2020, 2:56:49 AM] Error: The requested platform 'UnifiProtectMotion' was not registered by any plugin.
    at PluginManager.getPluginForPlatform (/usr/lib/node_modules/homebridge/src/pluginManager.ts:223:15)
    at /usr/lib/node_modules/homebridge/src/server.ts:386:37
    at Array.forEach (<anonymous>)
    at Server.loadPlatforms (/usr/lib/node_modules/homebridge/src/server.ts:373:27)
    at Server.start (/usr/lib/node_modules/homebridge/src/server.ts:152:29)
    at cli (/usr/lib/node_modules/homebridge/src/cli.ts:78:10)
    at Object.<anonymous> (/usr/lib/node_modules/homebridge/bin/homebridge:17:22)
    at Module._compile (internal/modules/cjs/loader.js:1156:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1176:10)
    at Module.load (internal/modules/cjs/loader.js:1000:32)
    at Function.Module._load (internal/modules/cjs/loader.js:899:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
    at internal/main/run_main_module.js:18:47
[5/13/2020, 2:56:49 AM] Got SIGTERM, shutting down Homebridge...
[5/13/2020, 2:56:54 AM] [HB Supervisor] Homebridge Process Ended. Code: 143, Signal: null

**Comment by beele** [comment]
Yeah I tried to fix the ARM build of tenstorflow which apparently broke something else, fixing now!

**Comment by beele** [comment]
If you try with the latest version (0.2.9) it 'should' work again.
The Tensorflow lib is giving my nightmares 😅

**Comment by nzbullet** [comment]
This is resolved for me however I was getting an Error: FFMPEG exited with code 1. I ended up changing some settings and it is working now. 

![3A165A60-6F54-490C-AD4A-2F4B56624271](https://user-images.githubusercontent.com/24196566/81777294-a2363e00-9544-11ea-92e5-c271af3f1f21.jpeg)

Just as an FYI:
**I removed**

- maxBitrate

**I changed**

- Lower case ‘p’ on _Person_ under enhanced_classes

- vcodec is set to h264 instead of h264_omx (on a RPi4)

This is my config: 
```
{
            "platform": "UnifiProtectMotion",
            "name": "Unifi Protect",
            "unifi": {
                "enhanced_classes": [
                    "Person"
                ],
                "controller": "https://xxx.xxx.xxx.xxx:7443",
                "controller_rtsp": "rtsp://xxx.xxx.xxx.xxx:7447",
                "username": "xxxxxxxxxxx",
                "password": "*************",
                "motion_interval": 20000,
                "motion_repeat_interval": 20000,
                "motion_score": 50,
                "enhanced_motion": true,
                "enhanced_motion_score": 50,
                "save_snapshot": false,
                "debug": false
            },
            "googlePhotos": {
                "upload_gphotos": false
            },
            "videoConfig": {
                "vcodec": "h264",
                "audio": true,
                "maxStreams": 2,
                "maxWidth": 1280,
                "maxHeight": 720,
                "maxFPS": 25,
                "mapvideo": "0:1",
                "mapaudio": "0:0",
                "packetSize": 188
            }
        }
```

**Comment by nzbullet** [comment]
Oh also FYI I am running Unifi Cloudkey G2 with 3 G3 Flex cams.

**Comment by beele** [comment]
@nzbullet the upper/lowercase 'Person' should not matter (it should be all lower case, I will fix that in my demo config). I convert it to lowercase in the code anyway.

I've seem to have tracked down the issue, it was a combination of me removing the step to rebuild the tfjs-lib (thinking it was not needed anymore) and missing dependencies (g++ on my linux test system).
I've added a new wiki page for node-gyp dependencies

**Comment by beele** [comment]
Latest version (0.3.1) should have this resolved.
@hfagelnour if you are still having issues check: https://github.com/beele/homebridge-unifi-protect-camera-motion/wiki/Trouble-installing-the-plugin%3F
and post a full install log

**Comment by beele** [comment]
Latest versions contains a number of fixes (0.3.3)

**Comment by beele** [comment]
This should be fixed, I cannot reproduce this myself anymore
